### PR TITLE
Adds JX querying functionality

### PR DESCRIPTION
**Pull request by nkremerh**
Adds JX fetch by URL or path
Adds JX select by boolean query
Adds JX project by arbitrary query
Adds JX schema retrieval
Adds JX object type-to-string
Fixes mislabeling of header file

Currently implemented functionality:
```
fetch(URL); || fetch(path);

select(bool-expr, [obj1, obj2, ...]);

project(expr, [obj1, obj2, ...]);

schema([obj1, obj2, ...]);
```

### CODE REVIEW

File: dttools/src/jx_querytools.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#include "jx_getopt.h"
+#include "jx_parse.h"
+#include "jx_print.h"
+#include "stringtools.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+struct jx * jx_fetch_from_url ( const char *u ) {
```
Comment (dthain) [inline dttools/src/jx_querytools.c:18]: Do these placeholders devolve to the jx function implementations for select, project, fetch, etc?

File: dttools/src/jx_function.c [hunk 2]
```diff
@@ -604,4 +608,192 @@ struct jx *jx_function_len(struct jx *args){
...
+	jx_delete(val);
+	return result;
+	
+	FAILURE:
+	jx_delete(args);
+	jx_delete(val);
+	FAIL(funcname, orig_args, err);
+}
+
+struct jx *jx_function_select(struct jx *orig_args, struct jx *ctx) {
```
Comment (dthain) [inline dttools/src/jx_function.c:691]: Check indenting throughout..

File: dttools/src/jx_function.c [hunk 3]
```diff
@@ -604,4 +608,192 @@ struct jx *jx_function_len(struct jx *args){
...
+	jx_delete(val);
+    return result;
+
+	FAILURE:
+	jx_delete(args);
+	jx_delete(val);
+	FAIL(funcname, orig_args, err);
+}
+
+struct jx *jx_function_schema(struct jx *orig_args, struct jx *ctx) {
```
Comment (dthain) [inline dttools/src/jx_function.c:754]: This looks interesting!  Any suggestions on how to use this effectively?

File: dttools/src/jx_eval.c [hunk 4]
```diff
@@ -333,12 +342,21 @@ static struct jx * jx_eval_operator( struct jx_operator *o, struct jx *context )
 	struct jx *right = NULL;
 	struct jx *result = NULL;
 
+    //Capture expression for select query before it gets evaluated
+	//Stringify it to prevent early evaluation
+    if(!strcmp("select", jx_print_string(o->left))) {
+		struct jx *r = jx_array_shift(o->right);
+		r = jx_string(jx_print_string((r)));
+		jx_array_insert(o->right, r);
+		fprintf(stderr, "inserting: %s\n", jx_print_string(o->right));
```
Comment (dthain) [inline dttools/src/jx_eval.c:351]: This looks like debug output the needs to be removed...

File: dttools/src/jx_querytools.c [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    size_t len = 0;
+    int flag;
+    while (getline(&line, &len, html) != -1) {
+        if(flag) return line;
+        if(string_match_regex(line, "<h1>Dumping raw contents of")) flag = 1;
+    }
+    return "";
+}
+
+struct jx * jx_query_project ( struct jx *c, struct jx *e ) {
```
Comment (dthain) [inline dttools/src/jx_querytools.c:34]: If this isn't called from anything, just remove it..

**Comment by nkremerh** [comment]
I believe all requested changes have been addressed.

**Pull request merged**